core/src/com/unciv/logic/civilization/managers/QuestManager.kt

```diff
@@ -896,23 +896,59 @@ class QuestManager : IsPartOfGameInfoSerialization {
 }
 
 
-class AssignedQuest(
-    val questName: String = "",
-    assigner: Civilization,
-    assignee: Civilization,
-    val assignedOnTurn: Int = 0,
-    val data1: String = "",
-    val data2: String = ""
-) : IsPartOfGameInfoSerialization {
-
-    val assigner = assigner.civName
+class AssignedQuest : IsPartOfGameInfoSerialization {
+
+    val questName: String
+    val assignedOnTurn: Int
+    val data1: String
+    val data2: String
+
+    constructor(
+        questName: String = "",
+        assigner: Civilization,
+        assignee: Civilization,
+        assignedOnTurn: Int = 0,
+        data1: String = "",
+        data2: String = ""
+    ) {
+        this.questName = questName
+        this.assignedOnTurn = assignedOnTurn
+        this.data1 = data1
+        this.data2 = data2
+        this.assigner = assigner.civName
+        this.assignerCiv = assigner
+        this.assignee = assignee.civName
+        this.assigneeCiv = assignee
+    }
+
+    @Suppress("unused") // Used in deserialization
+    @JvmOverloads
+    constructor(
```

Comment:
If the primary constructor is already no-args, when why this one? Also, `@JvmOverloads` is usually suspect and hints at conceptual mistakes.